src/components/posterior_component/posterior.sh

```diff
@@ -162,7 +178,13 @@ run_posterior() {
     LonMaxInvDomain=$(ncmax lon ${RunDirs}/StateVector.nc)
     LatMinInvDomain=$(ncmin lat ${RunDirs}/StateVector.nc)
     LatMaxInvDomain=$(ncmax lat ${RunDirs}/StateVector.nc)
-    nElements=$(ncmax StateVector ${RunDirs}/StateVector.nc ${OptimizeBCs})
+    nElements=$(ncmax StateVector ${RunDirs}/StateVector.nc)
+    if "$OptimizeBCs"; then
+	nElements=$((nElements+4))
+    fi
+    if "$OptimizeOH";then
+	nElements=$((nElements+1))
+    fi
```

Comment:
That's a good idea, but it would take some disentangling of the code so I suggest doing it as a separate pull request. 

Right now, we're utilizing this code in: 

1. Routine `setup_imi` within `src/components/setup_component/setup.sh`
2. Routine `setup_kf` within `src/components/kalman_component/kalman.sh`
3. Routine `reduce_dimension` within `components/statevector_component/statevector.sh`
4. Routine `run_posterior` within `components/posterior_component/posterior.sh`

The state vector file is currently created by calling `create_statevector` from within `setup_imi`. Moving the code for determining `nElements` out of `setup.sh` into `run_imi.sh` right after `setup_imi` is called would cause issues for routine `setup_jacobian` which uses `nElements` to determine how many run directories to create. One way around this is to remove the generation of the state vector file from `setup_imi` and into `run_imi.sh`. There may be other potential conflicts too, so we should think about how to do this more carefully.